src/imfilter.jl

```diff
@@ -6,18 +6,31 @@ function imfilter_cpu(img, krn)
   imfilter(img, centered(krn), Inner(), Algorithm.FFT())
 end
 
+# preallocated memory for krn optimization on GPU
+krnbuffer_gpu = nothing
```

Comment:
Hi @ClaroHenrique I think we are going in cycles here without actually addressing the issue. The implementation is not very clean and should instead start with a simple const global variable allocated once before the main loop in iqsim. I don't understand why we need an extra function just for that.

Reply:
The code was refactored as agreed in the discussion